WalletWasabi/Microservices/ProcessAsync.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Microservices
+{
+	/// <summary>
+	/// Async wrapper class for <see cref="System.Diagnostics.Process"/> class that implements <see cref="WaitForExitAsync(CancellationToken, bool)"/>
+	/// to asynchronously wait for a process to exit.
+	/// </summary>
+	/// <remarks><see cref="IDisposable"/> is implemented. Do not forget to use `using` or dispose any instance of this class.</remarks>
+	public class ProcessAsync : IDisposable
+	{
+		private readonly TaskCompletionSource<bool> TaskCompletionSource;
+		private readonly Process Process;
+
+		/// <summary>
+		/// To detect redundant calls.
+		/// </summary>
+		private bool _disposed = false;
+
+		public ProcessAsync(ProcessStartInfo startInfo) : this(new Process() { StartInfo = startInfo })
+		{
+		}
+
+		private ProcessAsync(Process process)
+		{
+			TaskCompletionSource = new TaskCompletionSource<bool>();
+
+			Process = process;
+			Process.EnableRaisingEvents = true;
+			Process.Exited += (object sender, EventArgs e) => TaskCompletionSource.TrySetResult(true);
+		}
+
+		/// <inheritdoc cref="Process.StartInfo"/>
+		public ProcessStartInfo StartInfo => Process.StartInfo;
+
+		/// <inheritdoc cref="Process.ExitCode"/>
+		public int ExitCode => Process.ExitCode;
+
+		/// <inheritdoc cref="Process.HasExited"/>
+		public bool HasExited => Process.HasExited;
+
+		/// <inheritdoc cref="Process.Id"/>
+		public int Id => Process.Id;
+
+		/// <inheritdoc cref="Process.StandardInput"/>
+		public StreamWriter StandardInput => Process.StandardInput;
+
+		/// <inheritdoc cref="Process.StandardOutput"/>
+		public StreamReader StandardOutput => Process.StandardOutput;
+
+		/// <inheritdoc cref="Process.Start()"/>
+		public void Start()
+		{
+			try
+			{
+				Process.Start();
+			}
+			catch
+			{
+				Logger.LogInfo($"{nameof(Process.StartInfo.FileName)}: {Process.StartInfo.FileName}");
+				Logger.LogInfo($"{nameof(Process.StartInfo.Arguments)}: {Process.StartInfo.Arguments}");
+				Logger.LogInfo($"{nameof(Process.StartInfo.RedirectStandardOutput)}: {Process.StartInfo.RedirectStandardOutput}");
+				Logger.LogInfo($"{nameof(Process.StartInfo.UseShellExecute)}: {Process.StartInfo.UseShellExecute}");
+				Logger.LogInfo($"{nameof(Process.StartInfo.CreateNoWindow)}: {Process.StartInfo.CreateNoWindow}");
+				Logger.LogInfo($"{nameof(Process.StartInfo.WindowStyle)}: {Process.StartInfo.WindowStyle}");
```

Comment:
Nit: add `.` at the end.

Reply:
This is a convention we follow for all the log entries. They have to end with a `.`.